tests/SoldierAPI.test.ts

```diff
@@ -0,0 +1,46 @@
+import request from 'supertest';
+import router from '../src/routes/routes';
+import Soldier from '../src/repository/soldierRepo';
+import * as soldiers from './testData';
+
+const app = router;
+beforeEach(async () => {
+  await Soldier.deleteMany();
+});
+describe('Insert Soldier into DB', () => {
+  it('Should Send an API Post Request and save soldier to DB, Expected 201:', async () => {
+    await request(app).post('/soldiers').send(soldiers.SoldierToBeAddedForPOSTtest).expect(201);
+  });
+});
+
+describe('Fail to insert Soldier into DB', () => {
+  it('Should Send an API Post Request and not save soldier to DB, Expected 400:', async () => {
```

Comment:
fix the test like I showed in the previous one... this one and all following tests